pkg/distributor/distributor.go

```diff
@@ -471,10 +471,11 @@ func (d *Distributor) divideEntriesBetweenShards(tenantID string, totalShards in
 
 func (d *Distributor) createShards(stream logproto.Stream, totalShards int, tenantID string, shardStreamsCfg *shardstreams.Config) ([]uint32, []streamTracker) {
 	var (
-		streamLabels   = labelTemplate(stream.Labels)
-		streamPattern  = streamLabels.String()
-		derivedKeys    = make([]uint32, 0, totalShards)
-		derivedStreams = make([]streamTracker, 0, totalShards)
+		streamLabels    = labelTemplate(stream.Labels)
+		streamPattern   = streamLabels.String()
+		derivedKeys     = make([]uint32, 0, totalShards)
+		derivedStreams  = make([]streamTracker, 0, totalShards)
+		entriesPerShard = int(math.Ceil(float64(len(stream.Entries)) / float64(totalShards)))
```

Comment:
by moving the `entriesPerShard` evaluation to this line you're doing it before the `totalShards <= 0` check, that is done at line 483. The check was added to guarantee against by 0 divisions.

That said, WDYT of moving the totalShards <= 0 to before the vars definition?